numpy/core/setup.py

```diff
@@ -1010,6 +1028,45 @@ def generate_umath_doc_header(ext, build_dir):
         # to make builds reproducible.
         svml_objs.sort()
 
+    highway_root = join('numpy', 'core', 'src', 'highway')
+    highway_sort = join(highway_root, 'hwy', 'contrib', 'sort')
+    highway_sort_objs = []
+    if can_link_highway():
+        highway_sort_objs = [
+            join(highway_root, 'hwy', 'aligned_allocator.cc'),
+            join(highway_root, 'hwy', 'per_target.cc'),
+            join(highway_root, 'hwy', 'print.cc'),
+            join(highway_root, 'hwy', 'targets.cc'),
+            *[
+                join(highway_sort, sort_file)
+                for sort_file in [
+                    "vqsort.cc",
+                    "vqsort_128a.cc",
+                    "vqsort_128d.cc",
+                    "vqsort_f32a.cc",
+                    "vqsort_f32d.cc",
+                    "vqsort_f64a.cc",
+                    "vqsort_f64d.cc",
+                    "vqsort_i16a.cc",
+                    "vqsort_i16d.cc",
+                    "vqsort_i32a.cc",
+                    "vqsort_i32d.cc",
+                    "vqsort_i64a.cc",
+                    "vqsort_i64d.cc",
+                    "vqsort_kv64a.cc",
+                    "vqsort_kv64d.cc",
+                    "vqsort_kv128a.cc",
+                    "vqsort_kv128d.cc",
+                    "vqsort_u16a.cc",
+                    "vqsort_u16d.cc",
+                    "vqsort_u32a.cc",
+                    "vqsort_u32d.cc",
+                    "vqsort_u64a.cc",
+                    "vqsort_u64d.cc",
+                ]
```

Comment:
This isn't ideal, when compiling without these files I got missing symbol issues, but we definitely don't use them all and they add ~100M to the .so.

@jan-wassenberg any thoughts on how to minimize the number of files we need here?

Reply:
I think we're only using {u/i/f}{32/64}a in your dispatcher and we ought to be able to remove 128*, 16*, and *d.cc (the latter for descending order).

If there are still linker errors after removing only those, then it's likely due to the dllexport. What we could then do is add a macro bitfield that you could define to replace the implementation of the unused/unwanted types with a stub.

Something like VQSORT_TYPES which contains multiple groups (one each for uint/int/float/KV), where each group has one bit encoding whether the element size equal to the bit's value is enabled.

Example, for only say unsigned/float types: `#define VQSORT_TYPES (((4 | 8) << 6) | (4 | 8))`.

Would that be useful?

Reply:
Sure, we can add that. As written, this will not support the 128-bit nor key-value types - is that OK?

Reply:
Right. And KV types also use different Traits.
Sounds good, we can indeed add that later.